Dynatrace/dynatrace-operator
# Set default topology spread constraints on activegate statefulset if none configured in dynakube.

# Description

Set topology spread constraints defaults in the statefulset of the ActiveGate. If the dynakube has topology spread constraints defined, they take precedence. Used default values:

```
  topologySpreadConstraints:
  - labelSelector:
      matchLabels:
        app.kubernetes.io/component: activegate
        app.kubernetes.io/name: activegate
    maxSkew: 1
    topologyKey: topology.kubernetes.io/zone
    whenUnsatisfiable: ScheduleAnyway
  - labelSelector:
      matchLabels:
        app.kubernetes.io/component: activegate
        app.kubernetes.io/name: activegate
    maxSkew: 1
    topologyKey: kubernetes.io/hostname
    whenUnsatisfiable: DoNotSchedule
```

## How can this be tested?
Deploy operator and dynakube once with configured constraints once without, observe statefulset.
Check how active gate pods are distributed on pods.


## Checklist
